src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInNameCheck.java

```diff
@@ -107,28 +122,33 @@
     private static final int DEFAULT_ALLOWED_ABBREVIATIONS_LENGTH = 3;
 
     /**
-     * Variable indicates on the allowed amount of capital letters in
-     * abbreviations in the classes, interfaces, variables and methods names.
+     * <b>Check Property</b>: indicates on the number of consecutive capital
+     * letters allowed in targeted identifiers (abbreviations in the classes,
+     * interfaces, variables and methods names, ... ).
      */
     private int allowedAbbreviationLength =
             DEFAULT_ALLOWED_ABBREVIATIONS_LENGTH;
 
     /**
-     * Set of allowed abbreviation to ignore in check.
+     * <b>Check Property</b>: list of abbreviations that must be skipped for checking.
+     * Abbreviations should be separated by comma.
      */
     private Set<String> allowedAbbreviations = new HashSet<>();
 
-    /** Allows to ignore variables with 'final' modifier. */
+    /** <b>Check Property</b>: allow to skip variables with {@code final} modifier. */
     private boolean ignoreFinal = true;
 
-    /** Allows to ignore variables with 'static' modifier. */
+    /** <b>Check Property</b>: allow to skip variables with {@code static} modifier. */
     private boolean ignoreStatic = true;
 
-    /** Allows to ignore methods with '@Override' annotation. */
+    /**
+     * <b>Check Property</b>: Allows to ignore methods tagged with {@code @Override} annotation
+     * (that usually mean inherited name).
+     */
     private boolean ignoreOverriddenMethods = true;
 
     /**
-     * Sets ignore option for variables with 'final' modifier.
+     * <b>Check Property Setter</b> : allow to skip variables with {@code final} modifier.
```

Comment:
`<b>Check Property Setter</b>` do we need single out setters ? what is a reason ?

Reply:
The original reason was to clearly state in code and javadoc that this is the setter for a check property.

Reply:
We do the same thing for properties themselves if you missed it.

Reply:
I noticed for properties.
At properties only it is ok to give signal to future generator of xdoc from javadoc that this description should be placed to table of properties. Why to both of them ?

It might be reasonable to use our special annotation over the field to make it clear that it is part of configuration.

Reply:
Like I said so it is clear incase we have some module where setter is not associated to property. `Checker.setModuleFactory` is one example.

> be reasonable to use our special annotation over the field to make it clear that it is part of configuration.

Yes, but this is another issue. This issue is mostly just about controlling the Javadocs. We can either enforce them to be something specific from xdoc, or be completely free-formed.
Will a annotation show up when the Javadocs are converted to HTML? If not, this is a reason to have the Javadoc text.

If they are not to be free-formed, I need to know what you want them to be forced to.

Reply:
@rnveach ,

>  we have some module where setter is not associated to property. Checker.setModuleFactory

it will always be ok, we will not mark this public method as `@PropertySetter`(or some other name of annotation).

> This issue is mostly just about controlling the Javadocs for a module.

as you bring this issue again, probably we need to come back to discussion on expected format.

> We can either enforce them to be something specific from xdoc, or let it be completely free-formed.

In my mind javadoc could be some formatted to be easily transformed to xdoc (checkstyle's future xdoc generator) and html format (javadoc tool from jdk). Focus will be more to xdoc, as most users read xdoc sourced documentation most of the time.

> Will a annotation show up when the Javadocs are converted to HTML?

html of javadoc - no
html (that come from xdoc) - it will let generator easily find what to put to xdoc.
Annotations is good for configuration.
Annotation should tell xdoc generator what it is nature of element/item and javadoc just contain html like contain of certain form for easy transformation to xdoc format of all text.
In other words: annotations are metadata, javadoc is collection of text values/descriptions.

Right now class field and filed setter have almost duplicated javadoc. Yes, is due to our obsession to apply to our code all that we produce. But it would be better to have javadoc only on setters (availbale to javadoc tool and xdoc generator).

Additionally annotations will help us to print all available properties for Module then smth unknown is found. To help user find typo on name or understand that property that he try to use is shown on web site, but it is available only from most recent version of checkstyle, and it does not exists in checkstyle version that is in use. It is still struggle for user to find out exact version of checkstyle they use(plugins, wrappers, .....). 

What I would do:
```
-    /** Allows to ignore variables with 'final' modifier. */
+    /** Allow to skip variables with {@code final} modifier. */
or
-    /** Allows to ignore variables with 'final' modifier. */
+    /** @see #setIgnoreFinal() */
```

and
```
-     * Sets ignore option for variables with 'final' modifier.
+     * Setter to allow to skip variables with {@code final} modifier.
...
@PropertySetter
```
xdoc will remove "Setter to"(or any other prefix) to use/expect in xdoc. So it mean when `@PropertySetter` is used, use first sentence of javadoc for content of xdoc property description. the same rule will be used to generate metadata xml file for eclipse-cs plugin or sonar plugin or ..... .
One more alternative, if you against of java anotation, we can use custom javadoc tag `@propertysetter`, no new types visible for users :) (in this case we loose ability to reuse metadata information in runtime, but we can use javadoc tags as first milestone and if it works well for us convert to java annotations)

Reply:
> html of javadoc - no
> if you against of java anotation

I am not against java annotation, my concern was the HTML version of the javadoc. When we started, we were concerned with how it would look, and I was just making a comment that you can't tell the difference between a property and a normal setter in HTML display. This is why I noted it in the Javadoc, so it would appear on the HTML display.

> +    /** @see #setIgnoreFinal() */

This is not possible as Javadoc is missing summary and our checks will complain.

> +    /** Allow to skip variables with {@code final} modifier. */
> +     * Setter to allow to skip variables with {@code final} modifier.

I will give this a try.